src/oInput/oInput.js

```diff
@@ -0,0 +1,98 @@
+import mapEvents from '../utils/eventMapper';
+import './input.css';
+import o from 'ojs-core';
+import generateRandomHash from '../utils/generateRandomHash';
+
+const TYPES = {
+    text: 'Text field',
+    date: 'Date field',
+    email: 'E-mail field',
+    month: 'Month field',
+    number: 'Number field',
+    password: 'Password field',
+    tel: 'Tel field',
+    file: 'File field',
+    url: 'URL field',
+};
+
+class oInput {
+    constructor(config) {
+        this.store = {
+            label: '',
+            type: 'text',
+            typeSpanText: false,
+            placeholder: '',
+            required: true,
+            index: false,
+            labelClass: 'input__label',
+            labelStyle: false,
+            inputClass: 'input__field',
+            inputStyle: false,
+            attributes: [],
+        };
+
+        this.configMerge(config);
+        this.store.id = config.id ? `${config.id}--input` : `${generateRandomHash()}--input`;
+    }
+
+    configMerge(config) {
```

Comment:
jak tak się już czepiam to może mergeConfig? :D Normalnie bym nie zwrócił na to uwagi, ale myślę, że i tak wrócisz do tego kodu, więc może akurat przy okazji :D Do przemyślenia.